modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -1352,6 +1355,65 @@ public void testSendNotificationToSubscribers() throws Exception {
 			objectEntryFolder1.getObjectEntryFolderId());
 	}
 
+	@FeatureFlag("LPD-50091")
+	@Test
+	public void testSendNotificationToUserGroups() throws Exception {
+		UserGroup userGroup1 = UserGroupTestUtil.addUserGroup();
+
+		_userGroupLocalService.addUserUserGroup(
+			user1.getUserId(), userGroup1.getUserGroupId());
+
+		UserGroup userGroup2 = UserGroupTestUtil.addUserGroup();
+
+		_userGroupLocalService.addUserUserGroup(
+			user1.getUserId(), userGroup2.getUserGroupId());
+
+		_userGroupLocalService.addUserUserGroup(
+			user2.getUserId(), userGroup2.getUserGroupId());
+
+		ObjectDefinition objectDefinition =
+			_addObjectDefinitionWithNotificationTemplateObjectAction(
+				NotificationRecipientConstants.TYPE_USER_GROUP,
+				userGroup1.getName(), userGroup2.getName());
+
+		Role guestRole = _roleLocalService.fetchRole(
+			TestPropsValues.getCompanyId(), RoleConstants.GUEST);
+
+		_resourcePermissionLocalService.setResourcePermissions(
+			TestPropsValues.getCompanyId(), objectDefinition.getClassName(),
+			ResourceConstants.SCOPE_COMPANY,
+			String.valueOf(TestPropsValues.getCompanyId()),
+			guestRole.getRoleId(), new String[] {ActionKeys.VIEW});
+
+		try {
+			_setUser(user1);
+
+			_testSendNotification(
+				null, StringPool.BLANK, user1, 1,
+				StringUtil.merge(
+					ListUtil.fromArray(
+						user1.getEmailAddress(), user2.getEmailAddress())),
+				ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+				objectDefinition);
+
+			objectDefinitionLocalService.deleteObjectDefinition(
+				objectDefinition);
+		}
+		finally {
+			_setUser(user2);
+
+			_userGroupLocalService.setUserUserGroups(
```

Comment:
Let’s mirror the approach used in UserNotificationTypeTest.java. Do we really need setUserUserGroups(...), or is deleteUserGroup(...) sufficient?

```suggestion
			_userGroupLocalService.setUserUserGroups(
				user1.getUserId(), new long[0]);
			_userGroupLocalService.setUserUserGroups(
				user2.getUserId(), new long[0]);

			_userGroupLocalService.deleteUserGroup(userGroup1);
			_userGroupLocalService.deleteUserGroup(userGroup2);
```